### DISCUSSION TITLE
Error: No tool releases found. package-lock.json

### DISCUSSION THREAD

**Discussion by tonnico** [renovatebot/renovate#17936] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.202.1

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitLab (.com and CE/EE)

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

Updating package-lock.json didn't work anymore. package.json is updated.
It worked with `32.149-slim` but stopped working with `>=32.150-slim`.

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Writing package-lock.json (repository=tests/renovate-test-project, branch=renovate/aws-cdk)
DEBUG: Massaging npm lock file before writing to disk (repository=tests/renovate-test-project, branch=renovate/aws-cdk)
DEBUG: Writing any updated package files (repository=tests/renovate-test-project, branch=renovate/aws-cdk)
DEBUG: Writing package.json (repository=tests/renovate-test-project, branch=renovate/aws-cdk)
DEBUG: npmrc file found in repository (repository=tests/renovate-test-project, branch=renovate/aws-cdk)
DEBUG: Generating package-lock.json for . (repository=tests/renovate-test-project, branch=renovate/aws-cdk)
DEBUG: Spawning npm install to create ./package-lock.json (repository=tests/renovate-test-project, branch=renovate/aws-cdk)
DEBUG: Updating lock file only (repository=tests/renovate-test-project, branch=renovate/aws-cdk)
DEBUG: Using node constraint ">= 16.14.2" from package.json (repository=tests/renovate-test-project, branch=renovate/aws-cdk)
DEBUG: Setting BUILDPACK_CACHE_DIR to /builds/tests/renovate-bot/renovate/cache/containerbase (repository=tests/renovate-test-project, branch=renovate/aws-cdk)
DEBUG: Using docker to execute (repository=tests/renovate-test-project, branch=renovate/aws-cdk)
       "image": "sidecar"
DEBUG: No concurrency limits (repository=tests/renovate-test-project, branch=renovate/aws-cdk)
       "host": "nodejs.org"
DEBUG: Datasource connection error (repository=tests/renovate-test-project, branch=renovate/aws-cdk)
       "datasource": "node",
       "packageName": "node",
       "url": undefined,
       "errCode": "EPROTO"
DEBUG: lock file error (repository=tests/renovate-test-project, branch=renovate/aws-cdk)
       "err": {
         "message": "No tool releases found.",
         "stack": "Error: No tool releases found.\n    at resolveConstraint (/usr/src/app/node_modules/renovate/lib/util/exec/containerbase.ts:182:11)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at generateInstallCommands (/usr/src/app/node_modules/renovate/lib/util/exec/containerbase.ts:233:27)\n    at prepareRawExec (/usr/src/app/node_modules/renovate/lib/util/exec/index.ts:131:11)\n    at exec (/usr/src/app/node_modules/renovate/lib/util/exec/index.ts:170:39)\n    at Object.generateLockFile (/usr/src/app/node_modules/renovate/lib/modules/manager/npm/post-update/npm.ts:124:5)\n    at getAdditionalFiles (/usr/src/app/node_modules/renovate/lib/modules/manager/npm/post-update/index.ts:545:17)\n    at processBranch (/usr/src/app/node_modules/renovate/lib/workers/repository/update/branch/index.ts:408:31)\n    at writeUpdates (/usr/src/app/node_modules/renovate/lib/workers/repository/process/write.ts:102:17)\n    at update (/usr/src/app/node_modules/renovate/lib/workers/repository/process/extract-update.ts:147:11)\n    at Object.renovateRepository (/usr/src/app/node_modules/renovate/lib/workers/repository/index.ts:52:19)\n    at Object.start (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:160:7)\n    at /usr/src/app/node_modules/renovate/lib/renovate.ts:16:22"
       },
       "type": "npm"
DEBUG: No updated lock files in branch (repository=tests/renovate-test-project, branch=renovate/aws-cdk)
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment]
Please run with v32, we don't support old versions

**Reply by tonnico** [reply]
I run it with v32.202.1. The description shows only the commit, with which it stopped working.

**Reply by mikaello** [reply]
> It worked with `21.149-slim` but stopped working with `>=21.150-slim`.

What do you mean by these versions, do you mean [`32.149`/`32.150`](https://github.com/renovatebot/renovate/compare/32.149.0...32.150.0)?

**Reply by tonnico** [reply]
Oh yes, my fault. @mikaello you are right. This is the correct commit.

**Comment by rarkins** [comment, marked as answer]
It tried to connect to https://nodejs.org and got this error:

```
DEBUG: Datasource connection error (repository=tests/renovate-test-project, branch=renovate/aws-cdk)
       "datasource": "node",
       "packageName": "node",
       "url": undefined,
       "errCode": "EPROTO"
```

https://www.google.com/search?q=nodejs+EPROTO indicates it might be an ipv4 vs ipv6 issue

**Reply by rarkins** [reply]
Could also be TLS version-related too

**Reply by tonnico** [reply]
@rarkins , this was a good hint. I need to add nodejs.org to my proxy allowlist.

Is it possible to add artifactory for this also?

**Reply by rarkins** [reply]
What do you mean by add artifactory?

**Reply by tonnico** [reply]
replace the download of the latest nodejs with an artifactory link.

**Reply by tonnico** [reply]
Ok, looks like, it is currently not possible.
https://github.com/containerbase/buildpack/blob/ca7bc48756551675f3ba518875d0138d98e2002f/src/usr/local/buildpack/tools/v2/node.sh#L50

**Reply by mikaello** [reply]
Replacing NodeJS.com with a proxy is discussed here: https://github.com/renovatebot/renovate/discussions/17874#discussioncomment-3688713

**Comment by mikaello** [comment]
I guess you are not running in an air gapped environment since you don't mention it?

But if you need a temporary workaround you can probably specify in your `package.json` the Node engine that is preinstalled in the Renovate slim image, [currently `v16.17.0`](https://github.com/renovatebot/docker-renovate/blob/f548870f8b74aabb7ac7c776abed47c327f21c9c/Dockerfile#L13-L14):

```json
  "engines": {
    "node": "16.17.0"
  },
```

Or as [I recently learned](https://github.com/renovatebot/renovate/discussions/17874#discussioncomment-3688713), setting env [`RENOVATE_BINARY_SOURCE=global`](https://docs.renovatebot.com/self-hosted-configuration/#binarysource), then Renovate will only use the pre bundled CLI tools, and not try to "dynamically" installing all tools on-demand.

**Reply by tonnico** [reply]
Not far away. I'm using `RENOVATE_BINARY_SOURCE=docker` and my package.json has the following:
```json
  "engines": {
    "node": ">= 16.14.2",
    "npm": ">= 8.5.0"
  }
```
So I would expect, that this container wouldn't upgrade the node version.
Removing this section didn't change the behavior.
